### ISSUE TITLE
Apigee emulator says "Datastore Error" when processing the RevokeOAuthV2 policy

### ISSUE THREAD

**Issue by toyopilgrim** [GoogleCloudPlatform/cloud-code-vscode#663]
Environment:

Apigee Runtime 1.8.0 (Apigee emulator)

Version information

Cloud Code Extension version: v1.21.0-insiders.1

VSCode version: 1.72.2

OS: Mac

Cloud SDK: Google Cloud SDK 392.0.0

Skaffold: N/A

Kubectl: N/A

Description:

After upgrading to 1.8.0 from 1.6.1, "Datastore Error" always occurs at RevokeOAuthV2

Exactly the same proxy did not have this issue when it was 1.6.1. 

Repro step:

With Apigee Emulator 1.8.0, execute the following policy given that this end user certainly has token in the KVM.

```
<RevokeOAuthV2 name="OAuthV2-RevokeToken">
  <EndUserId ref="myProxy.userId"/>
  <Cascade>true</Cascade>
</RevokeOAuthV2>
```

This an example of policy where I generate and store tokens :

```
<OAuthV2 name="OAuthV2-GenerateExternalAccessToken">
    <AppEndUser>myProxy.userId</AppEndUser>
    <ExpiresIn>600000</ExpiresIn>
    <ExternalAuthorization>true</ExternalAuthorization>
    <ExternalAccessToken>myProxy.accessToken</ExternalAccessToken>
    <GenerateResponse enabled="false"/>
    <Operation>GenerateAccessToken</Operation>
    <Scope>myProxy.scope</Scope>
    <StoreToken>true</StoreToken>
    <SupportedGrantTypes>
        <GrantType>client_credentials</GrantType>
    </SupportedGrantTypes>
</OAuthV2>
```

**Comment by ChaseMor** [comment]
Seems like a bug. We'll take a look!

**Comment by erica-addepar** [comment]
Bump on this bug :)

**Comment by shirishtrivikr** [comment]
@erica-addepar Thanks for reporting the issue.  This is being actively looked into. I will update the issue soon.

**Comment by progvb** [comment]
+1 for this.

I'd like to use emulator version 1.9.1 to refactor our logging to use the CloudLogging policy, since it has been fixed in #617 but I can't use it until this issue is resolved.

**Comment by shirishtrivikr** [comment]
@progvb 
I just verified that the Revoke access token policy works as expected.  Did you have chance to use the 1.9.1 emulator to verify that your usecase still fails ?

**Comment by progvb** [comment]
@shirishtrivikr Thank you, the problem doesn't seem to occur on my environment.

**Comment by progvb** [comment]
@shirishtrivikr 
Unfortunately, I'm getting this message:

```json
{"fault":{"faultstring":"Datastore Error","detail":{"errorcode":"Internal Server Error"}}}
```

**Comment by shirishtrivikr** [comment]
@progvb 
I was able to recreate the issue by using external token. Investigating further

**Comment by shirishtrivikr** [comment]
@progvb 
My execution went through when I changed Cascade to false.

**Comment by progvb** [comment]
@shirishtrivikr 
Mine doesn't even after setting Cascade to false:

```xml
<RevokeOAuthV2 name="OAuthV2-RevokeToken">
  <EndUserId ref="mygroup.userId"/>
  <Cascade>false</Cascade>
</RevokeOAuthV2>
```

**Comment by progvb** [comment]
@shirishtrivikr 
I've redeployed with setting Cascade to false and it went through on my machine, too.

```xml
<RevokeOAuthV2 name="OAuthV2-RevokeToken">
  <EndUserId ref="mygroup.userId"/>
  <Cascade>false</Cascade>
</RevokeOAuthV2>
```

**Comment by shirishtrivikr** [comment]
I just commented out the Cascade block and it worked for me

**Comment by shirishtrivikr** [comment]
This issue should be fixes now in the Emulator 1.9.2 [image](gcr.io/apigee-release/hybrid/apigee-emulator@sha256:69849f37d51030eaf8f1a2a560d03768de3bb3b12a8ee2772f2fd7f04e02860c).

**Comment by j-windsor** [comment]
Closing, as it looks like it is fixed. Thanks!
